### PR TITLE
uni_encoder

### PR DESCRIPTION
**Pull request by JaejinCho**
Please check if this could be merged

### CODE REVIEW

**Comment by sw005320** [comment]
Thanks!
This is a really valuable function
Could you remove some flake8 and pep8 errors in https://travis-ci.org/espnet/espnet/jobs/502299588#L1218-L1271 ?

Also, it's great if you report the performance difference from bidirectional RNN

File: espnet/nets/pytorch_backend/encoders.py [hunk 1]
```diff
@@ -24,19 +24,23 @@ class BRNNP(torch.nn.Module):
...
             else:
                 inputdim = hdim
-            rnn = torch.nn.LSTM(inputdim, cdim, dropout=dropout, num_layers=1, bidirectional=True,
-                                batch_first=True) if typ == "lstm" \
-                else torch.nn.GRU(inputdim, cdim, dropout=dropout, num_layers=1, bidirectional=True, batch_first=True)
-            setattr(self, "birnn%d" % i, rnn)
+            rnn = torch.nn.LSTM(inputdim, cdim, dropout=dropout, num_layers=1, bidirectional=bidir,
+                                batch_first=True) if "lstm" in typ \
+                else torch.nn.GRU(inputdim, cdim, dropout=dropout, num_layers=1, bidirectional=bidir, batch_first=True)
+            setattr(self, typ + "%d" % i, rnn)
```
Comment (kan-bayashi) [inline espnet/nets/pytorch_backend/encoders.py]: @sw005320 Do you want to keep model compatibility?
If so, we should use `birnn` or `rnn` for attibutes.

Comment (sw005320) [reply espnet/nets/pytorch_backend/encoders.py]: yes, let's do it. @JaejinCho, change it accordingly.

**Comment by pzelasko** [comment]
I think it would be good to rename `BRNN` -> `RNN` and `BRNNP` -> `RNNP` to avoid confusion (they can be unidirectional now).

**Comment by sw005320** [comment]
> I think it would be good to rename BRNN -> RNN and BRNNP -> RNNP to avoid confusion (they can be unidirectional now).

I agree.

@JaejinCho, could you modify the code accordingly based on @pzelasko and @kan-bayashi's comments?

**Comment by sw005320** [comment]
@JaejinCho, please fix the errors appeared in the travis-ci.
if you check https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md, you could test your changes by yourself.

**Comment by JaejinCho** [comment]
I found the chainer parts are not changed consistently for this update but I think it is necessary.
Can somebody familiar w/ chainer could fix it quickly? Or I can do it anyway.

**Comment by sw005320** [comment]
> I found the chainer parts are not changed consistently for this update but I think it is necessary.
Can somebody familiar w/ chainer could fix it quickly? Or I can do it anyway.

This is a good point. Maybe you can work with @Fhrozen? @Fhrozen, can you help @JaejinCho?
Also, it's better to have a test for this extension.
You could basically add unidirectional architecture types in https://github.com/espnet/espnet/blob/master/test/test_e2e_asr.py#L124-L161
I'll merge this PR, but it would be great if you prepare it as another PR.

**Pull request merged**